Tasks/AzureKeyVault/operations/KeyVault.ts

```diff
@@ -154,7 +171,11 @@ Set-AzureRmKeyVaultAccessPolicy -VaultName %s -ObjectId $spnObjectId -Permission
                     secretsToErrorsMap.addError(secretName, errorMessage);
                 }
                 else {
-                    this.setVaultVariable(secretName, secretValue);
+                    if (this.taskParameters.flattenVariableName) {
+                        this.flattenedSecrets[secretName] = secretValue;
```

Comment:
This might leak individual secret values - hence we need to register each secret value as secret so that value gets masked properly. Call trySetSecret inside flattenVariableName

Reply:
@lovakumar could you please have a look?

Reply:
Let's say you have the following variables in key vault.

"a" ="secret1"
"b" ="secret2"
"c"="secret3"

you configured FlattenVariableName to be "KeyVaultSecret".

you are expecting $KeyVaultSecret to be set to '{ "a" ="secret1", "b" ="secret2", "c"="secret3"}'.

Now let's say some other task emits/writes to log secret1 string. Since secret1 as value itself is a secret we expect it to be masked (i.e. print as *****).

In order to achieve above we will need to call trySetSecret function on each of the secret value.

If this isn't clear we can go over a skype meeting to discuss more on this.